.github/workflows/push-to-testnet-draft.yml

```diff
@@ -15,20 +15,28 @@ jobs:
         uses: actions/checkout@v2
       - name: Random sleep (balena API workaround)
         run:  sleep $[ ( $RANDOM % 180 )  + 1 ]s
-      - name: Change i2c bus for RockPi
+      - name: Generate Docker Compose for RockPi
         if: matrix.sbc == 'rockpi'
         run: |
-          sed -i -E "s/\/dev\/i2c-1:\/dev\/i2c-1/\/dev\/i2c-7:\/dev\/i2c-7/g" docker-compose.yml
+          python gen_docker_compose.py rockpi -o sample_output/rockpi.yml
+          rm docker-compose.yml
```

Comment:
not sure we need these `rm docker-compose.yml` bits anymore?

Reply:
Not sure if you have any particular opinion on this @marvinmarnold ? Re whether we need to keep the non-template docker compose

Reply:
@shawaj @SebastianMaj - Yes this looks like old code copied into the matrix execution. Since the file gets copied over on the [next line](https://github.com/NebraLtd/helium-miner-software/blob/kashif/use-dot-env-file/.github/workflows/push-to-testnet-draft.yml#L31) we can safely remote this line.

I don't like the `sample_output` folder name. That was before we integrated this into the github workflow. May be use a better name like `device_compose_files`?

> Not sure if you have any particular opinion on this @marvinmarnold ? Re whether we need to keep the non-template docker compose

I suggest keeping the generated docker-compose files for quick access for investigating possible issues.

Reply:
I agree, we should keep docker-compose.yml file as it's needed for testing locally but I was thinking we can maybe modify the gen_docker_compose.py so that it will write to root folder as docker-compose.yml if no output path is provided.

Reply:
Yes that can be done too. Though just passing `-o ./docker-compose.yml` would achieve the same thing?

Reply:
@kashifpk so you think we remove the `docker-compose.yml` in root, as I have done already, change `sample_output` folder to `device_compose_files` and then when those are generated as part of the github workflow, then commit them back to the repo each time and save the updated ones?

@SebastianMaj do you think it would be a major problem if we just had a folder with docker-compose-rockpi.yml and docker-compose-raspi.yml in there rather than just a single docker-compose.yml in the root of repo?

Reply:
> @kashifpk so you think we remove the docker-compose.yml in root, as I have done already, change sample_output folder to device_compose_files and then when those are generated as part of the github workflow, then commit them back to the repo each time and save the updated ones?

Yes.

Reply:
> @SebastianMaj do you think it would be a major problem if we just had a folder with docker-compose-rockpi.yml and docker-compose-raspi.yml in there rather than just a single docker-compose.yml in the root of repo?

I like the idea, I think it would be best that way. Then just as part of the github workflow we copy the file over to root depending on which step in the matrix we are at

Reply:
> balena to not complain lol

still surprised this isn't a feature of balena!!

Reply:
@kashifpk @SebastianMaj made some updates to the workflows as per suggestions here - so think should be ready for review now 👍